### PR TITLE
Report SQL query performance to Mixpanel

### PR DESCRIPTION
**Pull request by vinaysshenoy**
Context: https://simpledotorg.slack.com/archives/CFKD42741/p1620193199258200

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/storage/monitoring/AnalyticsReportingSqlDelegate.kt [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  }
+
+  private inline fun <reified R> runSqlOperationWithReporting(
+      daoInformation: DaoInformationExtractor.DaoMethodInformation,
+      operation: () -> R
+  ): R {
+    val operationStartTime = Instant.now(clock)
+    val result = operation()
+    val operationEndTime = Instant.now(clock)
+    val timeTakenForOperation = Duration.between(operationStartTime, operationEndTime).abs()
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/storage/monitoring/AnalyticsReportingSqlDelegate.kt]: There is a standard library function we can use to get the time taken to run an operation.

```kt
val timeTakenForOperation = measureTimeMillis { operation() }
```

**Review by msasikanth — APPROVED** [review]

File: app/src/main/java/org/simple/clinic/storage/monitoring/AnalyticsReportingSqlDelegate.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/storage/monitoring/AnalyticsReportingSqlDelegate.kt]: Interesting. Is this something new?

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/storage/monitoring/AnalyticsReportingSqlDelegate.kt]: Nope, there since Kotlin 1.0

**Pull request merged**